packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -124,13 +124,11 @@ function getPathInModule(
       const pkg = readPackageCached(packageJsonPath);
 
       if (pkg.exports) {
-        // we need to make sure resolve ignores `main`
-        delete pkg.main;
```

Comment:
we should not modify package cache content by `delete pkg.main`

Reply:
delete `pkg.main` of cache content will make the second `<rootDir>/packages/babel-jest` in `transform` fail because the missing main field. but I don't know why the failure is not happened before.